swift/Samples/AlertContainer/Sources/AlertContainerViewController.swift

```diff
@@ -0,0 +1,258 @@
+                    animations: {
+                        self.dimmingView.alpha = 1
+                        inAlertView.transform = .identity
+                        inAlertView.alpha = 1
+                    },
+                    completion: { _ in
+                        UIAccessibility.post(notification: .screenChanged, argument: nil)
+                    }
+                )
+            }
+        }
+        else {
+            if let alertView = alertView {
+                UIView.animate(
+                    withDuration: 0.1,
+                    delay: 0,
+                    options: .curveEaseInOut,
+                    animations: {
+                        alertView.transform = CGAffineTransform(scaleX: 0.85, y: 0.85)
+                        alertView.alpha = 0
+                        self.dimmingView.alpha = 0
+                    },
+                    completion: { _ in
+                        alertView.removeFromSuperview()
+                        self.view.setNeedsLayout()
+                        UIAccessibility.post(notification: .screenChanged, argument: nil)
+                    }
+                )
+                self.alertView = nil
+            }
+        }
+    }
+    
+    override var childForStatusBarStyle: UIViewController? {
+        return baseScreenViewController
+    }
+    
+    override var childForStatusBarHidden: UIViewController? {
+        return baseScreenViewController
+    }
+    
+    override var childForHomeIndicatorAutoHidden: UIViewController? {
+        return baseScreenViewController
+    }
+    
+    override var childForScreenEdgesDeferringSystemGestures: UIViewController? {
+        return baseScreenViewController
+    }
+    
+    public override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return baseScreenViewController.supportedInterfaceOrientations
+    }
+}
+
+private final class AlertView: UIView {
+    public var alert: Alert?
+    private lazy var title: UILabel = {
+        let title = UILabel()
+        title.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+        title.textAlignment = .center
+        title.translatesAutoresizingMaskIntoConstraints = false
+        return title
+    }()
+    
+    private lazy var message: UILabel = {
+        let message = UILabel()
+        message.font = UIFont.systemFont(ofSize: 18, weight: .medium)
+        message.textAlignment = .center
+        message.numberOfLines = 0
+        message.lineBreakMode = .byWordWrapping
+        message.translatesAutoresizingMaskIntoConstraints = false
+        return message
+    }()
+    
+    public required init(alert: Alert?) {
+        self.alert = alert
+        super.init(frame: .zero)
+        commonInit()
+    }
+    
+    public convenience override init(frame: CGRect) {
+        self.init(alert: nil)
+        self.frame = frame
+        commonInit()
+    }
+    
+    private func commonInit() {
+        if let alert = alert {
```

Comment:
A `guard` is a better choice here.

Reply:
In this case, there is else case since we are just not going to create the view alert is not nil. Alert in this case is the alert model not the alertView.  Am i missing something?

Reply:
Not sure I got your comment here. I was suggesting we use a `guard` here, like:

```swift
guard let alert = alert else {
  return
}
// Use alert here
```

`guard` can help with [code clarity](https://stackoverflow.com/a/38356532).

Reply:
there's a significant `else` clause so this might be awkward to use guard

```swift
guard let alert = alert else {
            if let alertView = alertView {
                UIView.animate(
                    withDuration: 0.1,
                    delay: 0,
                    options: .curveEaseInOut,
                    animations: {
                        alertView.transform = CGAffineTransform(scaleX: 0.85, y: 0.85)
                        alertView.alpha = 0
                        self.dimmingView.alpha = 0
                    },
                    completion: { _ in
                        alertView.removeFromSuperview()
                        self.view.setNeedsLayout()
                        UIAccessibility.post(notification: .screenChanged, argument: nil)
                    }
                )
                self.alertView = nil
            }
        }
    return
}
```

Reply:
Maybe I'm missing something, that's in a different spot, right? 🤔 https://github.com/square/workflow/blob/7d13724fdefbc3dbf8fb0b6dfb9291a3e21dad55/swift/Samples/AlertContainer/Sources/AlertContainerViewController.swift#L45

Reply:
ah yep your right, Github jumped to the wrong place while reviewing 👍

Reply:
@dhavalshreyas  got it! thanks for explaining 👍

Reply:
done.